pkg/reconciler/apiserversource/resources/receive_adapter.go

```diff
@@ -146,5 +146,15 @@ func makeEnv(args *ReceiveAdapterArgs) ([]corev1.EnvVar, error) {
 		Name:  "METRICS_DOMAIN",
 		Value: "knative.dev/eventing",
 	}}
-	return append(envs, args.Configs.ToEnvVars()...), nil
+
+	envs = append(envs, args.Configs.ToEnvVars()...)
+
+	if args.Source.Spec.CloudEventOverrides != nil && args.Source.Spec.CloudEventOverrides.Extensions != nil {
+		ceJson, err := json.Marshal(args.Source.Spec.CloudEventOverrides.Extensions)
+		if err != nil {
+			return nil, fmt.Errorf("Failure to marshal cloud event overrides %v: %v", args.Source.Spec.CloudEventOverrides.Extensions, err)
+		}
+		envs = append(envs, corev1.EnvVar{Name: "K_CE_OVERRIDES", Value: string(ceJson)})
```

Comment:
Sorry for my ignorance: Where does the ApiServerSource's image is picking up that environment variabel `K_CE_OVERRIDES` to add it to every CloudEvent it emits ?

Reply:
here: https://github.com/knative/eventing/blob/master/pkg/adapter/v2/config.go#L52
here: https://github.com/knative/eventing/blob/master/pkg/adapter/v2/main.go#L101
and here: https://github.com/knative/eventing/blob/master/pkg/adapter/v2/cloudevents.go#L83

Reply:
👍  thanks for the clarification